addons/azureSets/azuresets.lua

```diff
@@ -75,41 +75,90 @@ function set_spells(spellset)
         error(spellset..' was already equipped.')
         return
     end
-    windower.ffxi.reset_blue_magic_spells()
+    --[[windower.ffxi.reset_blue_magic_spells() ]]--
     log('Starting to set '..spellset..'.')
-    set_spells_from_spellset(spellset,1)
+    set_spells_from_spellset(spellset, 'remove')
     return
 end
 
-function set_spells_from_spellset(spellset,slot)
-    local islot
-    if tonumber(slot) < 10 then 
-        islot = '0'..slot
-    else islot = slot end
-    local tempname = settings.spellsets[spellset]['slot'..islot]
-    if tempname ~= nil then
-        for spell in spells:it() do
-                if spell['english']:lower() == tempname:lower() then
-                    windower.ffxi.set_blue_magic_spell(spell['id'], tonumber(slot))
-                    break
-                end
+function set_spells_from_spellset(spellset, setPhase)
+    local setToSet = settings.spellsets[spellset]
+    local currentSet = get_current_spellset()
+    --log_set(currentSet)
+    if setPhase == 'remove' then
+      -- Remove Phase
+      for k,v in pairs(currentSet) do
+        if not set_contains_value(setToSet, v) then
+          setSlot = k
+          local slotToRemove = tonumber(k:sub(5, k:len()))
+
+          windower.ffxi.remove_blue_magic_spell(slotToRemove)
+          --log('Removed spell: '..v..' at #'..slotToRemove)
+
+          windower.send_command('@wait .65;lua i azuresets set_spells_from_spellset '..spellset..' remove')
+          return
+        end
+      end
+    end
+    -- Did not find spell to remove. Start set phase
+    -- Find empty slot:
+    local slotToSetTo = -1
```

Comment:
Don't need to initialize this (it will be `nil` by default), then later check with `if not slotToSetTo then`.